crates/ingress-kafka/src/consumer_task.rs

```diff
@@ -457,36 +305,161 @@ impl Drop for AbortOnDrop {
     }
 }
 
-async fn topic_partition_queue_consumption_loop(
-    sender: MessageSender,
+struct TopicConsumptionTask<T, C>
```

Comment:
Maybe name `TopicPartitionConsumerTask` to denote that it reads from a single topic partition?